syntax = "proto3";

service Greeter {
    rpc SayHello(HelloRequest) returns (stream HelloReply) {}
    rpc SayRequestStream(stream HelloRequest) returns (HelloReply) {}
    rpc SayRequestAndRespStream(stream HelloRequest) returns (stream HelloReply) {}
}

message HelloRequest {
    string name = 1;
}

message HelloReply {
    string message = 1;
}